tests/test_poll_alerts.py

```diff
@@ -14,107 +15,172 @@
 )
 
 
-def test_poll_alerts():
+def poll_alerts(
```

Comment:
Why you have 2 definitions for `poll_alerts` (the other being in `skyportal_fink_client/skyportal_fink_client.py`)? It seems you could factorise code here.

Reply:
The issue is that I want to test the logic of that function, which means asserting things inside of it. So I simply copied it here and added some asserts. But if you know any way to like, "insert" some asserts in an already existing function it would help a lot !

Reply:
The reason why I'm testing this in the first place is just to try polling alerts right. Maybe the problem here is that the code to poll alerts and post them to SkyPortal in directly in that while loop, in this `poll_alerts` function. But what I should do maybe I write a separate function `poll_alert` that just polls one, to see if that works correctly. then a function that calls this one and then calls the function to post to SkyPortal, which is the function I would put in the while loop. That way I can test that poll+post function, and the main poll_alerts function I am just doing that in a loop which I believe doesn't need to be tested. What do you think?

Reply:
And if I need to test even the one with the "infinite" while loop, I can just add a "testing" parameter to it (already exists in the config to use a specific alert schema) that breaks the loop when there is no more alerts to poll.

Reply:
Yes, that sounds a good plan :-)